lib/src/client/connection.dart

```diff
@@ -13,11 +14,62 @@ class Connection {
   /// The stream socket subscription.
   final StreamSubscription<List<int>> _subscription;
 
+  /// Error handler from the latest listener.
+  void Function(Object, StackTrace) _onErrorListener;
+
+  /// Future that is resolved when the connection is closed.
```

Comment:
This should be part of README.md now

Reply:
I agree, we should also expose `done` from `Client`... It's slightly inaccessible right now.

But I'm also okay leaving error handling as a feature for advanced users for now. As long as we maintain the semantics that if you don't setup error handling an unhandled exception will occur. Then that exception will likely cause your isolate to crash and restart.